File: crossecore-java/src/java_/TypeDeclarationBase.java
package java_;
import org.eclipse.emf.common.notify.Notification;
import org.eclipse.emf.common.notify.NotificationChain;
import org.eclipse.emf.common.notify.impl.NotificationImpl;
import org.eclipse.emf.ecore.InternalEObject;
import org.eclipse.emf.ecore.*;
import org.eclipse.emf.ecore.impl.ENotificationImpl;
public class TypeDeclarationBase 
extends AbstractTypeDeclarationImpl implements TypeDeclaration
{
	private com.crossecore.ocl.OrderedSet<TypeParameter> _typeParameters;
	
	public com.crossecore.ocl.OrderedSet<TypeParameter> getTypeParameters()
	{
		if(_typeParameters==null){
			_typeParameters = new com.crossecore.ocl.OrderedSet<TypeParameter>(TypeParameter.class, this, Java_PackageImpl.TYPE_DECLARATION__TYPE_PARAMETERS, EOPPOSITE_FEATURE_BASE - Java_PackageImpl.TYPE_DECLARATION__TYPE_PARAMETERS);
		}
		return _typeParameters;
	
	}

	
	@Override
	protected EClass eStaticClass() {
		return Java_PackageImpl.Literals.TYPEDECLARATION;
	}
	
	
	
	@Override
	public Object eGet(int featureID, boolean resolve, boolean coreType) {
		switch (featureID) {
			case Java_PackageImpl.TYPE_DECLARATION__TYPE_PARAMETERS:
				return getTypeParameters();
		}
		return super.eGet(featureID, resolve, coreType);
	}
	
	@Override
	public void eSet(int featureID, Object newValue) {
		switch (featureID) {
			case Java_PackageImpl.TYPE_DECLARATION__TYPE_PARAMETERS:
				getTypeParameters().clear();
				getTypeParameters().addAll((java.util.Collection<? extends TypeParameter>) newValue);
				return;
		}
		super.eSet(featureID, newValue);
	}
	
	@Override
	public boolean eIsSet(int featureID) {
		switch (featureID) {
			case Java_PackageImpl.TYPE_DECLARATION__TYPE_PARAMETERS:
				return _typeParameters != null && !_typeParameters.isEmpty();
		}
		return super.eIsSet(featureID);
	}
	
	
	
}

File: crossecore-csharp/Java/Java/Java_/TypeDeclarationBase.cs
/* CrossEcore is a cross-platform modeling framework that generates C#, TypeScript, 
 * JavaScript, Swift code from Ecore models with embedded OCL (http://www.crossecore.org/).
 * The original Eclipse Modeling Framework is available at https://www.eclipse.org/modeling/emf/.
 * 
 * contributor: Simon Schwichtenberg
 */
 
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using oclstdlib;
using Ecore;
namespace Java_{
	public class TypeDeclarationBase 
	:AbstractTypeDeclarationImpl, TypeDeclaration
	{
		private OrderedSet<TypeParameter> _typeParameters;
		
		public virtual OrderedSet<TypeParameter> typeParameters
		{
			get {
				if(_typeParameters==null){
					_typeParameters = new OrderedSet<TypeParameter>(this, Java_PackageImpl.TYPEDECLARATION_TYPEPARAMETERS, EOPPOSITE_FEATURE_BASE - Java_PackageImpl.TYPEDECLARATION_TYPEPARAMETERS);
				}
				return _typeParameters;
			}
		
		}
		
		protected override EClass eStaticClass() {
			return Java_PackageImpl.Literals.TYPEDECLARATION;
		}
		
		
		
		public override object eGet(int featureID, bool resolve, bool coreType) {
			switch (featureID) {
				case Java_PackageImpl.TYPEDECLARATION_TYPEPARAMETERS:
					return typeParameters;
			}
			return base.eGet(featureID, resolve, coreType);
		}
		
		
		public override void eSet(int featureID, object newValue) {
			switch (featureID) {
				case Java_PackageImpl.TYPEDECLARATION_TYPEPARAMETERS:
					typeParameters.Clear();
					typeParameters.AddRange(((List<EObject>)newValue)?.Cast<TypeParameter>());
					return;
			}
			base.eSet(featureID, newValue);
		}
		
		/*
		public override bool eIsSet(int featureID) {
			switch (featureID) {
				case Java_PackageImpl.TYPEDECLARATION_TYPEPARAMETERS:
					return _typeParameters != null && !_typeParameters.isEmpty();
			}
			return base.eIsSet(featureID);
		}
		*/
		
		
		
	}
}
